langgraph/graph/message.py

```diff
@@ -1,18 +1,27 @@
 import uuid
-from typing import Annotated, TypedDict, Union
+from typing import Annotated, Literal, TypedDict, Union
 
 from langchain_core.messages import (
     AnyMessage,
     MessageLikeRepresentation,
     convert_to_messages,
     message_chunk_to_message,
 )
+from langchain_core.messages.base import BaseMessage
 
 from langgraph.graph.state import StateGraph
 
 Messages = Union[list[MessageLikeRepresentation], MessageLikeRepresentation]
 
 
+class DeleteMessage(BaseMessage):
```

Comment:
ooc: Why not just make this a named tuple or typed dict and update the add_messages signature?

Reply:
yea, it's ephemeral. no specific reason other than for consistency, i don't think it gives us any benefit (and might actually add more headache). let me see if i can make it work

Reply:
I had been doing this previously by having my reducer accept List[Union[AnyMessage, dict]] and then having the dict have an action -> IDK if we need to update our typing to let you differentiate between the channel type and the update type or anything though

Reply:
Seconding Will's suggestion if it's not massively more painful

Reply:
what are some other types of action just ooc? the obvious ones seem to be just `delete` / `update`, anything else?